### PR TITLE
[FIX] account_edi_ubl_cii: handle the date format while import invoices

### PR DESCRIPTION
**Pull request by arva-odoo**
When the user tries to import the invoice file with space on both sides of the date value the error will be generated.

Steps to reproduce:
1. Install the `account_edi_ubl_cii` module.
2. Generate one `factur-x.xml` file or import this file https://drive.google.com/file/d/1RCqLT37j2g6LPkauf75LpOzZt64XHd_L/view?usp=drive_link
3. Go to the invoice app and click the `upload` button.
4. Select the `factur-x.xml` file and an error will be generated at the backend and nothing to be imported.

Traceback in sentry: 
```
ValueError: time data '\n      20230601\n    ' does not match format '%Y%m%d'
  File "addons/account/models/account_move.py", line 3023, in _extend_with_attachments
    success = decoder(invoice, file_data, new)
  File "addons/account_edi_ubl_cii/models/account_edi_common.py", line 298, in _import_invoice_ubl_cii
    logs = self._import_fill_invoice_form(invoice, tree, qty_factor)
  File "addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py", line 315, in _import_fill_invoice_form
    date_obj = datetime.strptime(date_str, DEFAULT_FACTURX_DATE_FORMAT)
  File "_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
```
This commit fixes the issue by removing the extra spaces in the date value, if user gives the spaces in the invoice file.

see-
https://github.com/odoo/odoo/blob/a0ea6302a574624604cad7d0903711b30a4802a6/addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py#L315

sentry-4331059603

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by arva-odoo** [comment]
Hello @odoo/sentry_reviewers 
Please check this PR.
Thank you

File: addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py [hunk 1]
```diff
@@ -308,15 +309,15 @@ def _find_value(xpath, element=tree):
...
+            date_str = re.sub("[^0-9]", "", invoice_date_node.text)
             date_obj = datetime.strptime(date_str, DEFAULT_FACTURX_DATE_FORMAT)
             invoice_form.invoice_date = date_obj.strftime(DEFAULT_SERVER_DATE_FORMAT)
 
         # ==== invoice_date_due ====
 
         invoice_date_due_node = tree.find('.//{*}SpecifiedTradePaymentTerms/{*}DueDateDateTime/{*}DateTimeString')
         if invoice_date_due_node is not None and invoice_date_due_node.text:
-            date_str = invoice_date_due_node.text
+            date_str = re.sub("[^0-9]", "", invoice_date_due_node.text)
```
Comment (yelizariev) [inline addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py:320]: I'd put this patter to a new const and define it next to the definition of `DEFAULT_FACTURX_DATE_FORMAT`

Comment (arva-odoo) [reply addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py]: Changes are done!

File: addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py [hunk 2]
```diff
@@ -308,15 +310,15 @@ def _find_value(xpath, element=tree):
 
         invoice_date_node = tree.find('./{*}ExchangedDocument/{*}IssueDateTime/{*}DateTimeString')
         if invoice_date_node is not None and invoice_date_node.text:
-            date_str = invoice_date_node.text
+            date_str = re.sub(DEFAULT_FACTURX_DATE_PATTERN, "", invoice_date_node.text)
```
[the remark this answers is not in the document]
Comment (arva-odoo) [reply addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py]: Changes are done!

[the remark this answers is not in the document]
Comment (yelizariev) [reply addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py]: `DATE_REGEX` is neither good. When I see such a name I think that it has regexp for a date format, but in reality it has opposite meaning (regex for characters that should not be presented in date)

[the remark this answers is not in the document]
Comment (arva-odoo) [reply addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py]: Changes are done!

File: addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py [hunk 3]
```diff
@@ -308,15 +310,15 @@ def _find_value(xpath, element=tree):
 
         invoice_date_node = tree.find('./{*}ExchangedDocument/{*}IssueDateTime/{*}DateTimeString')
         if invoice_date_node is not None and invoice_date_node.text:
-            date_str = invoice_date_node.text
+            date_str = re.sub(CLEAR_DATE_REGEXP, "", invoice_date_node.text)
```
Comment (clbr-odoo) [inline addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py:313]:
```suggestion
            date_str = invoice_date_node.text.strip()
```
Just `strip()` :slightly_smiling_face: We don't wan't to remove everything but numbers here, if character or whatever but indentation ends up here we need to know something is wrong to fix it.

File: addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py [hunk 4]
```diff
@@ -308,15 +310,15 @@ def _find_value(xpath, element=tree):
...
             date_obj = datetime.strptime(date_str, DEFAULT_FACTURX_DATE_FORMAT)
             invoice_form.invoice_date = date_obj.strftime(DEFAULT_SERVER_DATE_FORMAT)
 
         # ==== invoice_date_due ====
 
         invoice_date_due_node = tree.find('.//{*}SpecifiedTradePaymentTerms/{*}DueDateDateTime/{*}DateTimeString')
         if invoice_date_due_node is not None and invoice_date_due_node.text:
-            date_str = invoice_date_due_node.text
+            date_str = re.sub(CLEAR_DATE_REGEXP, "", invoice_date_due_node.text)
```
Comment (clbr-odoo) [inline addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py:321]: Same. :point_up:

**Review by clbr-odoo — CHANGES_REQUESTED** [review]
Hello, thank you for the fix !
re.sub is overkill and may be problematic, I would just strip() the field :slightly_smiling_face:

**Comment by arva-odoo** [comment]
@smetl 
Changes are done!

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**